CODING_GUIDELINES.md

```diff
@@ -232,36 +232,61 @@ class MyClass
 ```
 
 ### 7. Header inclusion order. ###
-The headers should be placed in the following order:
- 1. Module header (in .cpp)
- 2. System/Qt/Boost etc. headers (split in subcategories if you have many).
- 3. Application headers, starting from *Base* headers.
+The headers should be placed in the following group order:
+  1. Module header (in .cpp)
+  2. C++ Standard Library headers
+  3. Boost library headers
+  4. Libtorrent headers
+  5. System headers
+  6. Qt headers
+  7. qBittorrent own headers, starting from *base* headers.
+
+The headers should be ordered alphabetically within each group.  
+If there are conditionals for header inclusion, put it at the bottom of the respective group.
```

Comment:
It confuses me. If several headers from different group are under the same condition, IMO, it's better to keep it together.

Reply:
>If several headers from different group are under the same condition, IMO, it's better to keep it together.

OK, I updated this part.